lib/bump/update_checkers/ruby.rb

```diff
@@ -22,23 +23,83 @@ def dependency_version
         # the Gemfile will appear in the Gemfile.lock. For instance, if a gem
         # specifies `platform: [:windows]`, and the Gemfile.lock is generated
         # on a Linux machine, the gem will be not appear in the lockfile.
-        parsed_lockfile.
-          specs.
+        parsed_lockfile.specs.
           find { |spec| spec.name == dependency.name }&.
           version
       end
 
       private
 
+      def fetch_latest_version
+        # If this dependency doesn't have a source specified we can just use
+        # Rubygems to get the latest version.
+        return Gems.info(dependency.name)["version"] if gem_remotes.none?
+
+        # Otherwise, we need to look at the versions in each of the specified
+        # sources. To start with, get an array of Bundler::Dependency::Fetchers
+        # for the remotes we need to look for this dependency at.
+        gem_source = Bundler::Source::Rubygems.new("remotes" => gem_remotes)
+        gem_fetchers =
+          gem_source.fetchers.flat_map(&:fetchers).
+          select { |f| f.is_a?(Bundler::Fetcher::Dependency) }
+
+        # Fetch the versions of this dependency available from each remote.
+        versions =
+          gem_fetchers.
+          flat_map { |f| f.unmarshalled_dep_gems([dependency.name]) }.
+          map { |details| Gem::Version.new(details.fetch(:number)) }
+
+        # Return the latest version that isn't a pre-release
+        versions.reject(&:prerelease?).sort.last.version
+      end
+
+      def gem_remotes
```

Comment:
This looks expensive, and we're calling it a few times in `#fetch_latest_version`. Memoize or assign to a local at the call site?